packages/adapter-cloudflare/README.md

```diff
@@ -83,7 +83,7 @@ declare namespace App {
 
 Functions contained in the `/functions` directory at the project's root will _not_ be included in the deployment, which is compiled to a [single `_worker.js` file](https://developers.cloudflare.com/pages/platform/functions/#advanced-mode). Functions should be implemented as [endpoints](https://kit.svelte.dev/docs/routing#endpoints) in your SvelteKit app.
 
-If you want to use `_headers` or `_redirects` custom [config files](https://developers.cloudflare.com/pages/platform/headers) to modify cloudflare behaviour you should put those configs in the `/static` folder of your SvelteKit project.
+The [`_headers` and `_redirects`](config files) files specific to Cloudflare Pages are not currently supported using SvelteKit. In order to return custom headers or redirect a user, return `headers` or a redirect using [SvelteKit Endpoints](https://kit.svelte.dev/docs/routing#endpoints).
```

Comment:
This is wrong. There is 2 different options. 
1. Cloudflare adapter for sites
2. Cloudflare adapter for workers

If you use cloudflare adapter for sites (new one) it actually supports headers and redirects, but they could be applied only for static content.

For example [here](https://securityheaders.com/?q=https%3A%2F%2Fmain.businessforukraine.pages.dev%2F&followRedirects=on) we added `x-robots-tag: noindex`

```_headers
/*
  X-Frame-Options: DENY
  X-Content-Type-Options: nosniff
  X-XSS-Protection: 1; mode=block
  Referrer-Policy: no-referrer-when-downgrade
  Permissions-Policy: document-domain=()

https://:businessforukraine.pages.dev/*
  X-Robots-Tag: noindex
  ```

all headers were added.

Reply:
Are you prerendering that page?

`env.ASSETS.fetch` will apply `_headers` and `_redirects` but `env.ASSETS.fetch` isn't used in SvelteKit SSR for the actual page itself (only images and static assets and the like). See:
https://github.com/sveltejs/kit/blob/2a9c19252912d61bd840441fe80dc13366af41f7/packages/adapter-cloudflare/src/worker.js#L20-L21

I agree we could leave this in but should be clarified that this is for static asset behavior only.

Reply:
Yep, we prerender most of content to get better performance. If you want to use shadowed endpoints or hooks for working with cookies this wouldn't work. But this is explicitly explained at cloudflare documentation for `_headers`

> Custom headers defined in the _headers file are not currently applied to responses from [Functions](https://developers.cloudflare.com/pages/platform/functions/), even if the function route matches the URL pattern. If your Pages application uses Functions, you must migrate any behaviors from the _headers file to the Response object in the appropriate /functions route. When altering headers for multiple routes, you may be interested in [adding middleware](https://developers.cloudflare.com/pages/platform/functions/#adding-middleware) for shared behavior.

from https://developers.cloudflare.com/pages/platform/headers/

This is the main benefit of new cloudflare adapter compared to old one. Previous one counted any request as worker invocation. New adapter allows to decrease workers consumption significantly.

Reply:
I'm afraid you're mistake on that: during beta, static assets are still counted as Functions calls.

But yeah, I saw that warning. We should specifically state this in the SvelteKit docs rather than linking to an external page however. I'll modify my PR.